src/lib/core/scroll/scroll-dispatcher.ts

```diff
@@ -0,0 +1,60 @@
+import {Injectable} from '@angular/core';
+import {Scrollable} from './scrollable';
+import {Subject} from 'rxjs/Subject';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/fromEvent';
+
+
+/**
+ * Service contained all registered Scrollable references and emits an event when any one of the
+ * Scrollable references emit a scrolled event.
+ */
+@Injectable()
+export class ScrollDispatcher {
+  /** Subject for notifying that a registered scrollable reference element has been scrolled. */
+  _scrolled: Subject<void> = new Subject<void>();
+
+  /**
+   * Map of all the scrollable references that are registered with the service and their
+   * scroll event subscriptions.
+   */
+  scrollableReferences: WeakMap<Scrollable, Subscription> = new WeakMap();
+
+  constructor() {
+    // By default, notify a scroll event when the document is scrolled or the window is resized.
+    Observable.fromEvent(window.document, 'scroll').subscribe(() => this._notify());
+    Observable.fromEvent(window, 'resize').subscribe(() => this._notify());
```

Comment:
Right now we are not prioritizing universal right now so this may be a concern for another time. Additionally I think there may be some confusion here as this is a service and not a component so it does not have access to a host nor have a lifecycle with destroy.